### PR TITLE
optimize the creation conditions of profiling task

### PR DESCRIPTION
**Pull request by xzyJavaX**
- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #[9742](https://github.com/apache/skywalking/issues/9742)
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/docs/en/changes/changes.md).

### CODE REVIEW

**Comment by wu-sheng** [comment]
Could you explain why removing the end-time condition helps?

**Comment by mrproliu** [comment]
Thanks for your contribution, I have two questions:
1. I don't think deleting the condition is a good idea, because we have other queries that need to use the end time bucket filter(such as cache timer). If you don't need the end time bucket filter, just set it as `null`.
2. If you ignore the end time bucket when creating a task, then the profiling will generally be created successfully(if the start time not be set), It's not right.

**Comment by wu-sheng** [comment]
I think we should build a strict condition to avoid overlap.

**Comment by mrproliu** [comment]
I think a good way to avoid overlap need following these steps:
1. find the recent profiling task from DB. An example SQL: `startBucket <= ${creating task end time}`.
2. Check the creating task start/end time is in the recent profiling task monitoring range.

**Comment by xzyJavaX** [comment]
I think this should be feasible:
Continue to judge after the alreadyHaveTaskList is found,if alreadyHaveTaskList isEmpty,continue to query DB,query condition is 
query.must(Query.range(ProfileTaskRecord.START_TIME).lte(endTimeBucket));
query.must(Query.range(ProfileTaskRecord.TIME_BUCKET).gte(endTimeBucket));
If the data can be found, it indicates that an overlay has occurred, and the task should not be created at this time.

**Comment by wu-sheng** [comment]
> Continue to judge after the alreadyHaveTaskList is found,if alreadyHaveTaskList isEmpty

This seems wrong from a grammar perspective. If a method named `alreadyHaveTaskList` return null, it should have no further query, right?

**Comment by wu-sheng** [comment]
I think you could `getProfileTaskDAO().getTaskList()`  to get a basic list, which should not be long, and just filter it in the memory.

**Comment by mrproliu** [comment]
> query.must(Query.range(ProfileTaskRecord.TIME_BUCKET).gte(endTimeBucket));

The `TIME_BUCKET` is usually the same as `START_TIME `, we should ignore it. 

> If the data can be found, it indicates that an overlay has occurred, and the task should not be created at this time.

You need to recheck again. If the previous task is already finished, the creation process should continue.

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/profiling/trace/ProfileTaskMutationService.java [hunk 1]
```diff
@@ -140,14 +140,16 @@ private String checkDataSuccess(final String serviceId,
...
-        long endTimeBucket = TimeBucket.getMinuteTimeBucket(monitorEndTime);
         final List<ProfileTask> alreadyHaveTaskList = getProfileTaskDAO().getTaskList(
-            serviceId, null, startTimeBucket, endTimeBucket, 1);
+            serviceId, null, startTimeBucket, null, null);
         if (CollectionUtils.isNotEmpty(alreadyHaveTaskList)) {
-            // if any task time bucket in this range, means already have task, because time bucket is base on task end time
-            return "current service already has monitor task execute at this time";
+            for (ProfileTask profileTask : alreadyHaveTaskList) {
+                if (profileTask.getStartTime() <= monitorEndTime) {
+                    // if any task time bucket in this range, means already have task, because time bucket is base on task end time
```
Comment (lujiajing1126) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/profiling/trace/ProfileTaskMutationService.java:148]:
```suggestion
                    // if the startTime is less or equal than the endTime of the newly created task, i.e. there is overlap between two tasks, it is an invalid case
```

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/profiling/trace/ProfileTaskMutationService.java [hunk 2]
```diff
@@ -140,14 +140,16 @@ private String checkDataSuccess(final String serviceId,
 
         // Each service can monitor up to 1 endpoints during the execution of tasks
         long startTimeBucket = TimeBucket.getMinuteTimeBucket(monitorStartTime);
-        long endTimeBucket = TimeBucket.getMinuteTimeBucket(monitorEndTime);
         final List<ProfileTask> alreadyHaveTaskList = getProfileTaskDAO().getTaskList(
-            serviceId, null, startTimeBucket, endTimeBucket, 1);
+            serviceId, null, startTimeBucket, null, null);
```
Comment (mrproliu) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/profiling/trace/ProfileTaskMutationService.java:144]: 1. If you use the start time to filter, then you may not get any data. For example, you create a 10 min task at 10:00, then when you create a new task at 10:05, then you cannot get the existing task. Am I right?
2. You should limit the profiling task count.

Comment (xzyJavaX) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/profiling/trace/ProfileTaskMutationService.java]: sorry, I used to think of TIME_BUCKE as END_TIME.
What do you think the "limit" should be set to.

Comment (xzyJavaX) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/profiling/trace/ProfileTaskMutationService.java]: sorry, I used to think of TIME_BUCKE as END_TIME.
What do you think the "limit" should be set to.

Comment (mrproliu) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/profiling/trace/ProfileTaskMutationService.java]: I think `1` is enough. Usually, you just need to check if the most recent task has overlap or not.

**Review by mrproliu — APPROVED** [review]
Thanks.

**Comment by wu-sheng** [comment]
3 ES relative profiling e2e fail. I just rerun to see whether they could pass, or there are some issues of this change.

**Pull request merged**